src/lib/KeyShares/KeyShares.ts

```diff
@@ -170,11 +171,12 @@ export class KeyShares {
    * Initialise from JSON or object data.
    */
   fromJson(content: string | any): KeyShares {
-    const data = typeof content === 'string'
-      ? JSON.parse(content).data
-      : content.data;
+    const body = typeof content === 'string' ? JSON.parse(content) : content;
+    if (!body.version || body.version !== `v${pkg.version}`) {
```

Comment:
This will work for minor+ and patch versions that are different, but it should raise exception for versions only minor+, patch versions should pass this check

Reply:
Also tests should be written properly to check these 2 cases

Reply:
So if tool is v1.0.1
Then keyshare of version v1.0.5 will be valid
and keyshare of versions v2.0.0 or v1.1.0 are invalid

Reply:
There is some package that brings good functionality to compare versions